src/common.rs

```diff
@@ -981,9 +982,78 @@ pub fn get_current_pid() -> Result<Pid, &'static str> {
     inner()
 }
 
+/// MAC address for network interface
+#[derive(PartialEq, Eq, Copy, Clone)]
+pub struct MacAddr {
+    data: [u8; 6],
+}
+
+impl MacAddr {
+    pub const UNSPECIFIED: Self = MacAddr::new();
+
+    #[inline]
+    pub(crate) const fn new() -> Self {
+        Self { data: [0; 6] }
+    }
+
+    pub fn data(&self) -> &[u8; 6] {
+        &self.data
+    }
+}
+
+impl From<[u8; 6]> for MacAddr {
+    fn from(data: [u8; 6]) -> Self {
+        Self { data }
+    }
+}
+
+impl FromStr for MacAddr {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let bytes = s
+            .split(':')
+            .filter_map(|s| u8::from_str_radix(s, 16).ok())
+            .collect::<Vec<u8>>();
+        if bytes.len() == 6 {
+            let mut data = [0; 6];
+            for (index, byte) in bytes.iter().enumerate() {
+                data[index] = *byte;
+            }
+            return Ok(MacAddr { data });
+        }
+        Err("invalid MAC address syntax".to_string())
+    }
+}
+
+impl std::fmt::Display for MacAddr {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let s = self
+            .data
+            .iter()
+            .map(|b| format!("{:02x}", *b))
+            .collect::<Vec<String>>()
+            .join(":");
+        f.write_str(&s)
+    }
+}
+
+// This type can hold different addresses associated to the network interface
+#[allow(unused)]
+pub(crate) enum InterfaceAddress {
+    // MAC address
+    MAC(MacAddr),
+    // IPv4 address and subnet mask
+    IPv4(Ipv4Addr, Ipv4Addr),
+    // Stub for IPv6
```

Comment:
Please turn this into a doc comment (useful when using `--document-private-items` in rustdoc).